pkg/operator/controller/ingress/status.go

```diff
@@ -123,21 +131,156 @@ func computeIngressAvailableCondition(deployment *appsv1.Deployment) operatorv1.
 	}
 }
 
-func computeIngressDegradedCondition(deployment *appsv1.Deployment) operatorv1.OperatorCondition {
+// computeDeploymentDegradedCondition computes the ingresscontroller's
+// "DeploymentDegraded" status condition by examining the status conditions of
+// the deployment.
+func computeDeploymentDegradedCondition(deployment *appsv1.Deployment) operatorv1.OperatorCondition {
 	for _, cond := range deployment.Status.Conditions {
-		if cond.Type == appsv1.DeploymentProgressing && cond.Status == corev1.ConditionFalse && cond.Reason == "ProgressDeadlineExceeded" {
-			return operatorv1.OperatorCondition{
-				Type:    operatorv1.OperatorStatusTypeDegraded,
-				Status:  operatorv1.ConditionTrue,
-				Reason:  "DeploymentFailed",
-				Message: fmt.Sprintf("The deployment failed (reason: %s) with message: %s", cond.Reason, cond.Message),
+		if cond.Type == appsv1.DeploymentAvailable {
+			switch cond.Status {
+			case corev1.ConditionFalse:
+				return operatorv1.OperatorCondition{
+					Type:    iov1.IngressControllerDeploymentDegradedConditionType,
+					Status:  operatorv1.ConditionTrue,
+					Reason:  "DeploymentAvailableFalse",
+					Message: fmt.Sprintf("The deployment has Available status condition set to False (reason: %s) with message: %s", cond.Reason, cond.Message),
+				}
+			case corev1.ConditionTrue:
+				return operatorv1.OperatorCondition{
+					Type:    iov1.IngressControllerDeploymentDegradedConditionType,
+					Status:  operatorv1.ConditionFalse,
+					Reason:  "DeploymentAvailableTrue",
```

Comment:
DeploymentAvailableTrue → DeploymentAvailable?

Reply:
Done.  I also changed DeploymentAvailableUnknown → DeploymentAvailabilityUnknown to enhance its prosaic similitude.